### PR TITLE
Change internal representation to i64

### PR DESCRIPTION
**Pull request by rinde**
This is a proposal to change the internal representation of both `Time` and `Duration` to `i64`. Not using `usize`/`isize` anymore [fixes a bug as the the library didn't work correctly on non-64 bits platforms](https://github.com/moia-oss/tinytime.rs/issues/4). The advantage of a uniform underlying type is that it eliminates many edge cases when converting between the types.

This PR has further increased clippy's strictness, which can be a very valuable safeguard, especially regarding casting.

An effect is that `Time` can now represent negative values. This can be used to represent values before 1970-01-01 or it could be used to count down to some arbitrary deadline (e.g. rocket launch in T-1h).

This PR does not yet fix / add test cases for negative `Time` values, such as the rounding functions. This should be addressed separately in follow-up PRs.

### CODE REVIEW

**Review by lxwbr — APPROVED** [review]

**Review by jankeu — APPROVED** [review]
lgtm

File: src/lib.rs [hunk 1]
```diff
@@ -34,33 +34,32 @@ use serde::Serialize;
...
 #[derive(
     Eq, PartialEq, Hash, Ord, PartialOrd, Copy, Clone, Debug, Default, Serialize, Deref, From, Into,
 )]
-pub struct Time(usize);
+pub struct Time(i64);
 impl Time {
-    pub const MAX: Self = Self(usize::MAX);
-    pub const EPOCH: Self = Self(0_usize);
+    pub const MAX: Self = Self(i64::MAX);
+    pub const EPOCH: Self = Self(0);
```
Comment (jankeu) [inline src/lib.rs:44]: We probably want to either set Time::default() to Time::EPOCH, or set Time::EPOCH to i64::MIN. Otherwise a lot of test cases in route-optimizer will likely fail.
